### PR TITLE
Implement ADAr writer

### PR DESCRIPTION
**Pull request by SkyToGround**
### Description of work

See title.

### Reminder

*Changes should be documented in `changes.md`*

### CODE REVIEW

File: src/AccessMessageMetadata/ADAr/ADAr_Extractor.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace AccessMessageMetadata {
+using FlatbufferMessage = FileWriter::FlatbufferMessage;
+using FBReaderBase = FileWriter::FlatbufferReader;
+
+/// See parent class for documentation.
+class ADAr_Extractor : public FBReaderBase {
+public:
+  bool verify(FlatbufferMessage const &Message) const override;
+  std::string source_name(FlatbufferMessage const &) const override;
+  uint64_t timestamp(FlatbufferMessage const &Message) const override;
```
Comment (matthew-d-jones) [inline src/AccessMessageMetadata/ADAr/ADAr_Extractor.h:28]: These 3 could all be marked `nodiscard`

File: src/AccessMessageMetadata/ADAr/ADAr_Extractor.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+/// \file
+/// \brief Define classes required to implement the ADC file writing module.
+
+#pragma once
+
+#include "FlatbufferReader.h"
+
+namespace AccessMessageMetadata {
+using FlatbufferMessage = FileWriter::FlatbufferMessage;
+using FBReaderBase = FileWriter::FlatbufferReader;
```
Comment (matthew-d-jones) [inline src/AccessMessageMetadata/ADAr/ADAr_Extractor.h:21]: `using FBReaderBase` seems overkill for one use of the class, but obviously not a big deal.

File: src/WriterModule/ADAr/ADAr_Writer.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  void config_post_processing() override;
+
+  InitResult init_hdf(hdf5::node::Group &HDFGroup) override;
+
+  InitResult reopen(hdf5::node::Group &HDFGroup) override;
+
+  void write(FlatbufferMessage const &Message) override;
+
+  static std::uint64_t epicsTimeToNsec(std::uint64_t sec, std::uint64_t nsec);
```
Comment (matthew-d-jones) [inline src/WriterModule/ADAr/ADAr_Writer.h:42]: Not implemented?

**Review by matthew-d-jones — CHANGES_REQUESTED** [review]
I have not manually tested with ADAr data from Kafka but implementation lgtm, a couple of minor comments below. I suggest it is worth adding a system test once there is support in streaming-data-types library.

File: src/AccessMessageMetadata/ADAr/ADAr_Extractor.h (hunk 2 above)
Comment (SkyToGround) [reply src/AccessMessageMetadata/ADAr/ADAr_Extractor.h:21]: Fair comment, I think it is the result of this PR largely being copy paste code.

File: src/WriterModule/ADAr/ADAr_Writer.h (hunk 3 above)
Comment (SkyToGround) [reply src/WriterModule/ADAr/ADAr_Writer.h:42]: Right, another result of copy-paste.

**Review by matthew-d-jones — APPROVED** [review]

**Pull request merged**